File: MinNetIOCP/FirstPersonController.lua
local gameObject;
local component;

function RPC(methodName, target, ...)

	local paramCount = select("#", ...);
	local param = MinNetPool.PopPacket();
	param:create_packet()

	for i = 1, paramCount do

		param:push(select(i, ...));

	end

	gameObject:sendRPC(methodName, target, param);

end

function SyncPosition (packet)

	--print("SyncPosition");

	local position = packet:pop_vector3 ();
	local euler = packet:pop_vector3 ();
	local cameraEuler = packet:pop_vector3 ();
	
	gameObject.position = position;

end

--[[function SetGameObject (object)
	--print("오브젝트 설정");
	gameObject = object
end]]--

function SetComponent (object)

	component = object;
	gameObject = object.gameObject;

end

File: MinNetIOCP/FirstPersonController.cpp
#include "FirstPersonController.h"


FirstPersonController::~FirstPersonController()
{
}

void FirstPersonController::InitRPC()
{
	DefRPC("SyncPosition", std::bind(&FirstPersonController::SyncPosition, this));
}

void FirstPersonController::SyncPosition()
{
	auto position = rpcPacket->pop_vector3();
	auto euler = rpcPacket->pop_vector3();
	auto cameraEuler = rpcPacket->pop_vector3();

	gameObject->position = position;

	std::cout << gameObject->position << std::endl;

}
